test/security/StanleyPauseManager.t.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.16;
+
+import "forge-std/Test.sol";
+import "../../contracts/security/PauseManager.sol";
+import "../../contracts/vault/StanleyUsdc.sol";
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "../../contracts/mocks/tokens/MockTestnetToken.sol";
+import "../../contracts/tokens/IvToken.sol";
+import "../../contracts/mocks/stanley/MockTestnetStrategy.sol";
+
+contract StanleyPauseManagerTest is Test {
+    address private _owner;
+    address private _user1;
+    address private _user2;
+
+    MockTestnetToken private usdc = new MockTestnetToken("Mocked USDC", "USDC", 100_000_000 * 1e6, uint8(6));
+    IvToken ivUsdc = new IvToken("IV USDC", "ivUSDC", address(usdc));
+
+    function setUp() public {
+        _owner = vm.rememberKey(1);
+        _user1 = vm.rememberKey(2);
+        _user2 = vm.rememberKey(3);
+    }
+
+    function testShouldNotPauseIfNoGuardianIsSet() public {
+        // given
+        Stanley stanley = createStanley();
+
+        // when & then
+        vm.startPrank(_user1);
+        vm.expectRevert(abi.encodePacked("IPOR_011"));
+        stanley.pause();
+    }
+
+    function testShouldNotPauseWhenCalledByNonGuardian() public {
+        // given
+        Stanley stanley = createStanley();
+        vm.prank(_owner);
+        stanley.addGuardian(_user1);
+
+        // when & then
+        vm.startPrank(_user2);
+        vm.expectRevert(abi.encodePacked("IPOR_011"));
+        stanley.pause();
+    }
+
+    function testShouldPauseWhenCalledByGuardian() public {
+        // given
+        Stanley stanley = createStanley();
+        assertFalse(stanley.paused());
+        vm.prank(_owner);
+        stanley.addGuardian(_user1);
+
+        // when
+        vm.startPrank(_user1);
+        stanley.pause();
+
+        // then
+        assertTrue(stanley.paused());
+    }
+
+    function testShouldNotPauseWhenCalledByRemovedGuardian() public {
+        // given
+        Stanley stanley = createStanley();
+        assertFalse(stanley.paused());
+        vm.prank(_owner);
+        stanley.addGuardian(_user1);
+
+        // when
+        vm.prank(_owner);
+        stanley.removeGuardian(_user1);
+
+        // then
+        vm.startPrank(_user1);
+        vm.expectRevert(abi.encodePacked("IPOR_011"));
+        stanley.pause();
+        assertFalse(stanley.paused());
+    }
+
+    function testShouldNotRemoveGuardianWhenCalledByNonOwner() public {
+        // given
+        Stanley stanley = createStanley();
+
+        // when & then
+        vm.startPrank(_user2);
+        vm.expectRevert(abi.encodePacked("Ownable: caller is not the owner"));
+        stanley.removeGuardian(_user1);
+    }
+
+    function testShouldNotAddGuardianWhenCalledByNonOwner() public {
```

Comment:
this can be done in `test/security/PauseManager.t.sol`

Reply:
I cannot add these tests here because the library has no owner